ra2ce/analysis/adaptation/adaptation_option.py

```diff
@@ -72,23 +72,25 @@ def from_config(
         """
         if (
             not analysis_config.config_data.damages_list
-            or not analysis_config.config_data.losses_list
+            and not analysis_config.config_data.losses_list
         ):
             raise ValueError(
-                "Damages and losses sections are required to create an adaptation option."
+                "Damages and/or losses sections are required to create an adaptation option."
             )
 
-        # Create input for the analyses
+        # Create input for the analyses (if given in config)
         _analyses = [
             AdaptationOptionAnalysis.from_config(
                 analysis_config=analysis_config,
-                analysis_type=_analysis,
+                analysis_type=_analysis_type,
                 option_id=adaptation_option.id,
             )
-            for _analysis in [
+            for _analysis_type in [
                 AnalysisDamagesEnum.DAMAGES,
                 analysis_config.config_data.adaptation.losses_analysis,
             ]
+            if _analysis_type
+            in (x.analysis for x in analysis_config.config_data.analyses)
```

Comment:
I'm not entirely sure how this works. Do we generate the iterator `(x.analysis for x in analysis_config.config_data.analyses)` for every step of the `for-loop` ? Or is it just created once? In case of unclear answer I would suggest declaring the available analyses outside the `for-loop`.

Reply:
It will be generated multiple times. Refactored to build it outside the loop.